### PR TITLE
Constant check for `coeFill`, `coeInv` and `coeInvFill`

### PR DESCRIPTION
**Pull request by imkiva**
This PR also changed signature of `coeFill` and `coeInvFill` to:
- `coeFill (A : I -> Type) (phi : I) : Pi (u : A 0) -> Path A u (coe A phi u)`
- `coeInvFill (A : I -> Type) (phi : I) : Pi (u : A 1) -> Path (\i => A (~ i)) u (coeInv A phi u)`

so they are desugared in the same way as `coe` and `coeInv`:
- `coe (A : I -> Type) (phi : I) : A 0 -> A 1`
- `coeInv (A : I -> Type) (phi : I) : A 1 -> A 0`

### CODE REVIEW

File: base/src/main/java/org/aya/core/def/PrimDef.java [hunk 1]
```diff
@@ -310,6 +394,10 @@ public enum ID {
     ID(@NotNull String id) {
       this.id = id;
     }
+
+    public static boolean isCoercion(@NotNull ID id) {
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/def/PrimDef.java:398]:
```suggestion
    public static boolean projSyntax(@NotNull ID id) {
```

**Comment by ice1000** [comment]
What

**Comment by ice1000** [comment]
@imkiva do you still have the code? I don't know that happened and why the pr was closed and I accidentally hit delete branch omg

**Comment by imkiva** [comment]
> @imkiva do you still have the code? I don't know that happened and why the pr was closed and I accidentally hit delete branch omg

Restored. Maybe a bug of GitHub or space, as I see:

image

File: base/src/main/java/org/aya/core/def/PrimDef.java [hunk 2]
```diff
@@ -96,6 +99,70 @@ private final class Initializer {
...
+      public final @NotNull PrimDef.PrimSeed coeFill = new PrimSeed(ID.COEFILL, this::coeFill, ref -> {
+        // coeFill (A : I -> Type) (phi : I) : Pi (u : A 0) -> Path A u (coe A phi u)
+        return coeFillFactory(ref, ID.COEFILL, ID.COE, FormulaTerm.LEFT);
+      }, ImmutableSeq.of(ID.I, ID.COE));
+
+      private @NotNull Term coeFill(@NotNull PrimCall prim, @NotNull TyckState state) {
+        var type = prim.args().get(0).term();
+        var restr = prim.args().get(1).term();
+        var j = new LocalVar("j");
+        return CoeTerm.coeFill(type, isOne(restr), new RefTerm(j));
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/def/PrimDef.java:111]: Are you sure this is correct?

**Review by ice1000 — COMMENTED** [review]
I recommend moving the computation of `coeInv` to `CoeTerm`

**Comment by ice1000** [comment]
Is this ready to be merged? Looks quite good

**Comment by imkiva** [comment]
> Is this ready to be merged? Looks quite good

I am adapting some agda code as our tests.

File: base/src/main/java/org/aya/core/def/PrimDef.java (hunk 2 above)
Comment (imkiva) [reply base/src/main/java/org/aya/core/def/PrimDef.java:111]: Oh! sorry!

**Comment by ice1000** [comment]
Merge?

**Comment by imkiva** [comment]
bors d=ice1000

**Comment by ice1000** [comment]
virs r+

**Pull request merged**